lightning/src/routing/router.rs

```diff
@@ -1512,6 +1515,80 @@ where L::Target: Logger {
 		}
 	}
 
+	// Step (10).
+	// Add a random 'shadow route' offset to the CLTV expiry delta values to improve privacy.
+	for path in selected_paths.iter_mut() {
+		let mut shadow_ctlv_expiry_delta_offset: u32 = 0;
+
+		// Choose the last publicly known node as the starting point for the random walk
+		if let Some(starting_hop) = path.iter().rev().find(|h| network_nodes.contains_key(&NodeId::from_pubkey(&h.as_ref().unwrap().pubkey))) {
+			let mut cur_node_id = NodeId::from_pubkey(&starting_hop.as_ref().unwrap().pubkey);
+
+			// Init PRNG with path nonce
+			let mut path_nonce = [0u8; 8];
+			path_nonce.copy_from_slice(&cur_node_id.as_slice()[..8]);
+			let mut prng = ChaCha20::new(random_seed_bytes, &path_nonce);
+			let mut random_path_bytes = [0u8; 8];
+
+			// Pick a random path length in [1 .. 3]
+			prng.process_in_place(&mut random_path_bytes);
+			let random_walk_length = usize::from_be_bytes(random_path_bytes).wrapping_rem(3).wrapping_add(1);
```

Comment:
I believe this will fail to compile on 32-bit platforms. Probably want a u64::from_be_bytes instead (and below) and then convert to usize.

Reply:
Ow, that's a good point. However, as Rust even supports `usize`s down to `u16` and squeezing a `u64` into a 32 bit `usize` wouldn't be lossless anyways, I now opted to simply use a `random_bytes` array of size `::core::mem::size_of::<usize>()` in https://github.com/lightningdevkit/rust-lightning/pull/1286/commits/ea06afb642fa3fc02aad1c9ad7466da7498cccc7.